lib/dataset/test/bins_reduction_test.cpp

```diff
@@ -0,0 +1,186 @@
+};
+
+TEST_F(DataArrayBinsNaNReductionTest, nansum) {
+  EXPECT_EQ(bins_nansum(binned_da), DataArray(makeVariable<double>(
+                                        indices.dims(), Values{1, 0, 0, 9})));
+}
+
+TEST_F(DataArrayBinsNaNReductionTest, nanmax) {
+  EXPECT_EQ(
+      bins_nanmax(binned_da),
+      DataArray(makeVariable<double>(
+          indices.dims(), Values{1.0, std::numeric_limits<double>::lowest(),
+                                 std::numeric_limits<double>::lowest(), 5.0})));
+}
+
+TEST_F(DataArrayBinsNaNReductionTest, nanmin) {
+  EXPECT_EQ(
+      bins_nanmin(binned_da),
+      DataArray(makeVariable<double>(
+          indices.dims(), Values{1.0, std::numeric_limits<double>::max(),
+                                 std::numeric_limits<double>::max(), 4.0})));
+}
+
+TEST_F(DataArrayBinsNaNReductionTest, nanmean) {
+  const auto res = bins_nanmean(binned_da);
+  EXPECT_EQ(res.slice({Dim::Y, 0}),
+            DataArray(makeVariable<double>(Dims{}, Values{1.0})));
+  EXPECT_TRUE(isnan(res.slice({Dim::Y, 1})).data().value<bool>());
+  EXPECT_TRUE(isnan(res.slice({Dim::Y, 2})).data().value<bool>());
+  EXPECT_EQ(res.slice({Dim::Y, 3}),
+            DataArray(makeVariable<double>(Dims{}, Values{4.5})));
+}
+
+class DataArrayBoolBinsReductionTest : public ::testing::Test {
+protected:
+  Dimensions dims{Dim::Y, 4};
+  Variable indices = makeVariable<scipp::index_pair>(
+      dims, Values{std::pair{0, 2}, std::pair{2, 2}, std::pair{2, 3},
+                   std::pair{4, 6}});
+  Variable data = makeVariable<bool>(
+      Dims{Dim::X}, Shape{6}, Values{true, false, false, true, true, true});
+  DataArray buffer = DataArray(data);
+  Variable binned_var = make_bins(indices, Dim::X, copy(buffer));
+  DataArray binned_da{binned_var};
+};
+
+TEST_F(DataArrayBoolBinsReductionTest, sum) {
+  EXPECT_EQ(bins_sum(binned_da),
+            DataArray(makeVariable<int64_t>(indices.dims(), Values{1, 0, 0, 2},
+                                            units::none)));
+}
+
+TEST_F(DataArrayBoolBinsReductionTest, max) {
+  EXPECT_EQ(bins_max(binned_da),
+            DataArray(makeVariable<bool>(indices.dims(),
+                                         Values{true, false, false, true})));
+}
+
+TEST_F(DataArrayBoolBinsReductionTest, min) {
+  EXPECT_EQ(bins_min(binned_da),
+            DataArray(makeVariable<bool>(indices.dims(),
+                                         Values{false, true, false, true})));
+}
+
+TEST_F(DataArrayBoolBinsReductionTest, all) {
+  EXPECT_EQ(bins_all(binned_da),
+            DataArray(makeVariable<bool>(indices.dims(),
+                                         Values{false, true, false, true})));
+}
+
+TEST_F(DataArrayBoolBinsReductionTest, any) {
+  EXPECT_EQ(bins_any(binned_da),
+            DataArray(makeVariable<bool>(indices.dims(),
+                                         Values{true, false, false, true})));
+}
+
+class DataArrayBinsMaskedReductionTest : public ::testing::Test {
```

Comment:
Any reason for not testing the `nan*` variants here?

Reply:
No. I was unsure how much testing they need at this level. But I can easily add some tests.

Reply:
Since masks and NaNs have a 'similar' effect (they get ignored) we should test that also the combination works.